client/extensions/woocommerce/woocommerce-services/views/carrier-accounts/dynamic-settings.js

```diff
@@ -0,0 +1,66 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+
+/**
+ * Internal dependencies
+ */
+import * as api from 'woocommerce/woocommerce-services/api';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import DynamicCarrierAccountSettingsForm from './dynamic-settings-form';
+
+export const DynamicCarrierAccountSettings = ( props ) => {
+	const [carrierRegistrationFields, setCarrierRegistrationFields] = useState([]);
+
+	useEffect(() => {
+		const fetchRegistrationFields = async () => {
+			const registrationFields = await api.get(props.siteId, api.url.shippingCarrierTypes());
+			setCarrierRegistrationFields(registrationFields.carriers);
+		}
+		fetchRegistrationFields();
+	}, [props.siteId]);
+
+	if (!carrierRegistrationFields || carrierRegistrationFields.length < 1) {
+		return (
+			<div>Loading...</div>
+		);
+	}
+
+	const [ currentCarrierRegistrationField ] = carrierRegistrationFields.filter(carrier => carrier.type === props.carrier);
+
+	if (!currentCarrierRegistrationField) {
+		return (
+			<div>{props.carrier} not supported.</div>
+		);
+	}
+
+    return (
+		<div>
+			<DynamicCarrierAccountSettingsForm
+				carrierType={currentCarrierRegistrationField.type}
+				carrierName={currentCarrierRegistrationField.name}
+				registrationFields={currentCarrierRegistrationField.fields}
+			/>
+		</div>
+	);
+};
+
+const mapStateToProps = ( state ) => {
+	return {
+		siteId: getSelectedSiteId( state ),
+    };
+};
+
+DynamicCarrierAccountSettings.propTypes = {
+	carrier: PropTypes.string.isRequired,
+	siteId: PropTypes.number
+};
+
+export default connect( mapStateToProps )( localize( DynamicCarrierAccountSettings ) );
```

Comment:
The props injected by the `localize` HOC are not used by this component
```suggestion
export default connect( mapStateToProps )( DynamicCarrierAccountSettings );
```

Reply:
Good call. We should localizing. I am going to wrap text with `translate`.